src/ps.rs

```diff
@@ -0,0 +1,303 @@
+// Copyright 2015-2016, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use std::ffi::CString;
+use std::ops::Deref;
+use std::path::Path;
+use std::io;
+
+use ffi;
+use ffi::enums::{SurfaceType, PsLevel};
+use surface::{Surface, SurfaceExt};
+use support;
+
+#[cfg(feature = "use_glib")]
+use glib::translate::*;
+
+pub struct File {
```

Comment:
I see some potential of sharing code here between PDF and PS. It's basically the same except for the types and the functions called.